src/Vault.sol

```diff
@@ -835,8 +852,8 @@ contract Vault {
     @return success Boolean indicating if there is sufficient free margin to increase the margin position
     @dev All values expressed in the base currency of the vault with same number of decimals as the base currency.
     */
-    function increaseMarginPosition(uint256 baseCurrency, uint256 amount) public onlyAuthorized returns (bool success) {
-        if (baseCurrency != vault._baseCurrency) _setBaseCurrency(baseCurrency);
+    function increaseMarginPosition(address baseCurrency, uint256 amount) public onlyAuthorized returns (bool success) {
+        if (baseCurrency != vault.baseCurrency) _setBaseCurrency(baseCurrency);
         success = getFreeMargin() >= amount;
```

Comment:
If there is only one protocol that can call this function, why not have "protocol.openposition" included as input in this function? As it's currently implemented this requires another roundtrip

Reply:
Because in the future there should be more functions that can call this function